dashboard_register.go

```diff
@@ -167,6 +167,7 @@ func (h *HTTPDashboardHandler) sendHeartBeat() error {
 	req := h.newRequest(h.HeartBeatEndpoint)
 	req.Header.Set("x-tyk-nodeid", NodeID)
 	req.Header.Set("x-tyk-nonce", ServiceNonce)
+	req.Close = true
```

Comment:
I think this change will eliminate keep-alived connections and increase latency for tyk->dashboard requests which seems to be not what we want to do.

I think that bug might be here below: https://github.com/TykTechnologies/tyk/pull/1952/files#diff-eca768c13a66fc0ccb769e695e73f949R174

We do `defer resp.Body.Close()` only if it was reply 200, for other 3XX, 4XX, 5XX replies we don't do it and introduce connection leak in clients pool of connections.